src/TTImp/Elab/Record.idr

```diff
@@ -233,3 +234,84 @@ checkUpdate rig elabinfo nest env fc upds rec expected
                 rcase <- recUpdate rig elabinfo fc nest env upds rec recty'
                 log "elab.record" 5 $ "Record update: " ++ show rcase
                 check rig elabinfo nest env rcase expected
+
+export
+elabInstance : {vars : _} ->
+                {auto c : Ref Ctxt Defs} ->
+                {auto m : Ref MD Metadata} ->
+                {auto u : Ref UST UState} ->
+                {auto e : Ref EST (EState vars)} ->
+                Env Term vars ->
+                FC -> Name -> List IFieldUpdate ->
+                Core RawImp
+elabInstance env fc n fs
+    = do
+         defs <- get Ctxt
+         names@(_ :: _) <- getConNames defs n
+              | _ => throw (NotRecordType fc n)
+         names@(_ :: _) <- mapMaybe (sequenceM . bimap pure id)
+                              <$> (sequence $ map (sequenceCore . bimap pure (findFieldsExplicit defs) . dup) names)
```

Comment:
This is really hard to understand. Can we not have something simpler using
`flip traverse names $ \ n => do (...)`?